Source/WebKit/Configurations/Base.xcconfig

```diff
@@ -120,6 +120,17 @@ SWIFT_VERSION = 5.0;
 OTHER_SWIFT_FLAGS = $(inherited) $(OTHER_SWIFT_FLAGS$(WK_MACOS_1400)) -no-warnings-as-errors -Xfrontend -experimental-spi-only-imports -Xfrontend -enable-experimental-concurrency -Xfrontend -enable-upcoming-feature -Xfrontend IsolatedDefaultValues $(OTHER_SWIFT_FLAGS_AVAILABILITY_$(USE_INTERNAL_SDK));
 OTHER_SWIFT_FLAGS_AVAILABILITY_YES = @$(WK_WEBKITADDITIONS_HEADERS_FOLDER_PATH)/Scripts/availability-definitions.txt;
 
+SWIFT_ACTIVE_COMPILATION_CONDITIONS = $(inherited) $(SWIFT_ACTIVE_COMPILATION_CONDITIONS_$(WK_PLATFORM_NAME));
+SWIFT_ACTIVE_COMPILATION_CONDITIONS_iphoneos = $(SWIFT_ACTIVE_COMPILATION_CONDITIONS$(WK_IOS_18));
+SWIFT_ACTIVE_COMPILATION_CONDITIONS_iphonesimulator = $(SWIFT_ACTIVE_COMPILATION_CONDITIONS$(WK_IOS_18));
+SWIFT_ACTIVE_COMPILATION_CONDITIONS_xros = $(SWIFT_ACTIVE_COMPILATION_CONDITIONS$(WK_IOS_18));
+SWIFT_ACTIVE_COMPILATION_CONDITIONS_xrsimulator = $(SWIFT_ACTIVE_COMPILATION_CONDITIONS$(WK_IOS_18));
+SWIFT_ACTIVE_COMPILATION_CONDITIONS_IOS_SINCE_18 = ENABLE_SWIFTUI;
+SWIFT_ACTIVE_COMPILATION_CONDITIONS_maccatalyst = $(SWIFT_ACTIVE_COMPILATION_CONDITIONS_maccatalyst$(WK_MACOS_1500));
+SWIFT_ACTIVE_COMPILATION_CONDITIONS_maccatalyst_MACOS_SINCE_1500 = ENABLE_SWIFTUI;
+SWIFT_ACTIVE_COMPILATION_CONDITIONS_macosx = $(SWIFT_ACTIVE_COMPILATION_CONDITIONS$(WK_MACOS_1500));
+SWIFT_ACTIVE_COMPILATION_CONDITIONS_MACOS_SINCE_1500 = ENABLE_SWIFTUI;
```

Comment:
I think you can safely check $(WK_MACOS_1500) and $(WK_IOS_18) without checking WK_PLATFORM_NAME first. Unless you found otherwise?

Reply:
I need it to not be true on watchOS or visionOS, which is what happened when I stuck with just the Mac and iOS pieces